test/unit/interface_merge_test.rb

```diff
@@ -0,0 +1,58 @@
+require 'test_helper'
+
+class InterfaceMergeTest < ActiveSupport::TestCase
+  def setup
+    @merge = InterfaceMerge.new
+    @attributes = compute_attributes(:with_interfaces)
+    @profile_name = @attributes.compute_profile.name
+  end
+
+  def attrs(num)
+    {
+      'attr' => num,
+      'from_profile' => @profile_name
+    }
+  end
+
+  test "it merges compute attributes with existing NICs" do
+    interfaces = [
+      Nic::Managed.new(:identifier => 'eth0'),
+      Nic::Managed.new(:identifier => 'eth1'),
+      Nic::Managed.new(:identifier => 'eth2')
+    ]
+    @merge.run(interfaces, @attributes)
+
+    interfaces.length.must_equal 3
+    interfaces[0].compute_attributes.must_equal(attrs(1))
+    interfaces[0].identifier.must_equal('eth0')
+
+    interfaces[1].compute_attributes.must_equal(attrs(2))
+    interfaces[1].identifier.must_equal('eth1')
+
+    interfaces[2].compute_attributes.must_equal({})
+    interfaces[2].identifier.must_equal('eth2')
+  end
+
+  test "it creates NICs when there aren't any" do
+    interfaces = []
+    @merge.run(interfaces, @attributes)
+
+    interfaces.length.must_equal 2
+    interfaces[0].compute_attributes.must_equal(attrs(1))
+    interfaces[1].compute_attributes.must_equal(attrs(2))
+  end
+
+  test "it creates additional NICs" do
+    interfaces = [
+      Nic::Managed.new(:identifier => 'eth0')
+    ]
+    @merge.run(interfaces, @attributes)
+
+    interfaces.length.must_equal 2
+    interfaces[0].compute_attributes.must_equal(attrs(1))
+    interfaces[0].identifier.must_equal('eth0')
+
+    interfaces[1].compute_attributes.must_equal(attrs(2))
+    interfaces[1].identifier.must_equal(nil)
+  end
```

Comment:
Setting profile name is covered by comparing with `attrs`. See line 10 of the test. I'll rename the method to `expected_attrs` to make it more obvious.